### PR TITLE
DSD-1155: Add requiredLabelText to Label and TextInput components

### PR DESCRIPTION
**Pull request by isastettler**
Fixes JIRA ticket [DSD-1155](https://jira.nypl.org/browse/DSD-1155)

## This PR does the following:

- Adds requiredLabelText prop to TextInput and Label Component
- Adds tests for TextInput and Label 
- Updates the jest-test-results file

## Accessibility concerns or updates

-

### Checklist:

- [x] I have updated the Storybook documentation accordingly.
- [ ] I have added relevant accessibility documentation for this pull request.
- [x] All new and existing tests passed.

### Front End Review:

- [ ] Review the Vercel preview deployment once it is ready.

### CODE REVIEW

**Review by EdwinGuzman — CHANGES_REQUESTED** [review]
The code update itself looks good but add context for the new prop in `Label.stories.mdx`. It should be a new heading after the existing `isRequired helper text` heading.

This new section should state that the `requiredLabelText` prop is available to use to update the `" (Required)"` text but should be used carefully. Only text that means "requires", including in other languages, is recommended. You can also say we don't recommended more than one or two words for the text.

**Review by EdwinGuzman — COMMENTED** [review]
Almost forgot, add this update in the CHANGELOG and update the version in `Label.stories.mdx` and `TextInput.stories.mdx` to `1.2.0`. @bigfishdesign13 I'd like to get this out soon because it's needed for the Library Card App.

File: src/components/Label/Label.stories.mdx [hunk 1]
```diff
@@ -31,14 +31,15 @@ import { getCategory } from "../../utils/componentCategories";
...
-| Latest            | `1.0.6`    |
+| Latest            | `1.2.0`    |
 
 ## Table of Contents
 
 - [Overview](#overview)
 - [Component Props](#component-props)
 - [Accessibility](#accessibility)
 - [isRequired helper text](#isrequired-helper-text)
+- [requiredLabelText](#requiredlabeltext-to-customize-helper-text)
```
Comment (EdwinGuzman) [inline src/components/Label/Label.stories.mdx]: The text inside the brackets `[]` should be the same as the hash inside the parenthesis `()`. Let's rename the text to `requiredLabelText to customize required label`. The "helper text" is already a concept for form components and saying the same here might be confusing.

File: src/components/Label/Label.stories.mdx [hunk 2]
```diff
@@ -98,3 +99,28 @@ by default.
     </VStack>
   </Story>
 </Canvas>
+
+## requiredLabelText to customize helper text
+
+The requiredLabelText prop accepts values fot type string and is available to update the "(Required)" text.
+It is recommend to be used carefully and to only pass text that means "required", including in other languages.
+We also recommend a maximum of two words for this text.
+Note: The arenthesis will be added automatically by the component.
```
Comment (EdwinGuzman) [inline src/components/Label/Label.stories.mdx]: This is good but some updates:
- add backticks to `requiredLabelText`
- typo "fot" should be "for"
- typo "arenthesis" should be "parenthesis"

File: src/components/Label/Label.stories.mdx [hunk 3]
```diff
@@ -98,3 +99,28 @@ by default.
...
+
+<Canvas>
+  <Story name="requiredLabelText">
+    <VStack>
+      <Label htmlFor="label-id1" id="regular" isRequired>
+        A regular isRequired helper text
```
Comment (EdwinGuzman) [inline src/components/Label/Label.stories.mdx]: Change "helper text" to "label".

File: src/components/Label/Label.stories.mdx (hunk 2 above)
Comment (isastettler) [reply src/components/Label/Label.stories.mdx]: fixed

File: src/components/Label/Label.stories.mdx (hunk 1 above)
Comment (isastettler) [reply src/components/Label/Label.stories.mdx]: changed

File: src/components/Label/Label.stories.mdx (hunk 3 above)
Comment (isastettler) [reply src/components/Label/Label.stories.mdx]: done

**Review by EdwinGuzman — APPROVED** [review]
Don't merge yet until @bigfishdesign13 gets a chance to review.

**Review by bigfishdesign13 — APPROVED** [review]
This looks good.  I am glad you added the note about the parentheses being added by the component.  There would have been (and probably still will be) a LOT of questions about that.

**Comment by bigfishdesign13** [comment]
@EdwinGuzman Will you handle the merge into the PR for the `rc`?  Sorry, I did not realize this was inrended for the `1.2.0` release.  I would have jumped right on this.

**Comment by EdwinGuzman** [comment]
Yeah, I'll handle the merge.

**Pull request merged**